### ISSUE TITLE
Feature request: social login

### ISSUE THREAD

**Issue by dragGH102** [NativeScript/NativeScript#1298]
On Ionic / Cordova it's very easy to find a "login with Google/Facebook etc" plugin.
However I didn't find any for Nativescript.

I think it would be extremely handy.

**Comment by N3ll** [comment]
Hi @dragGH102 , for Facebook login, you check this [plugin](https://www.npmjs.com/package/nativescript-facebook-login).
And about the Google login- we are currently working on it and we will inform you as soon as we are ready. 
Hope this is helpful! :)

**Comment by dragGH102** [comment]
thank you.
will this also allow me to use custom share (I. e. share on a specific
social network from the app, not the "share to" menu)?

also, do you plan to release similar plug-ins for other social network
(Instagram, twitter,..)?

thank you and keep up the good work.

**Comment by sitefinitysteve** [comment]
Just leaving this here as well:
https://www.npmjs.com/package/nativescript-auth0
https://auth0.com/why-auth0

**Comment by dragGH102** [comment]
On Wed, Jan 6, 2016 at 5:28 PM, Steve McNiven-Scott <
notifications@github.com> wrote:

> https://www.npmjs.com/package/nativescript-auth0


thanks @Steve.

just to make sure I understood correctly, this won't work on Android as of
now. right?

**Comment by sitefinitysteve** [comment]
@dragGH102 Yeah, not a limitation of the plugin, limitation of {N} on android right now.  What will make it work on android looks to maybe be out for 1.6.
https://github.com/NativeScript/NativeScript/issues/1330

**Comment by dragGH102** [comment]
On Wed, Jan 6, 2016 at 6:04 PM, Steve McNiven-Scott <
notifications@github.com> wrote:

> looks to maybe be out for 1.6.


yes looks so. Then I assume the plugin will just work without modifications
once the {N} issue is solved.
Thanks for the advice


-- 
*Manuel Gajo*
Full-stack web Developer and Business Designer
*www.manuelgajo.me <http://www.manuelgajo.me>*
+39 333 623 4217 / +32 484 369 265

*IMPORTANT*: My e-mail is personal and PRIVATE and CAN'T be shared with any
third party company or organisation without my explicit authorisation.

**Comment by valentinstoychev** [comment]
Just to confirm that we treat this issue with high importance, so if there are no unexpected circumstances should be done for 1.6, which is coming in a month.

**Comment by jlooper** [comment]
That's great news!!!

**Comment by hdeshev** [comment]
Here's a weird idea for doing social logins: port the [ng-cordova-auth](https://github.com/nraboy/ng-cordova-oauth) plugin by showing a WebView and destroying it after the user grants access, when you get your hands on the API token.

It's not 100% native, pure, and everything, but it will be **very** easy to set up. In addition people will be able to use all the cool auth providers like Untappd. :)

**Comment by dragGH102** [comment]
I would greatly appreciate full compatibility for a native plugin but i think webview itself is a very good idea.

Perhaps (as a temporary solution) we can use WebView on Android as documented at : http://docs.nativescript.org/ApiReference/ui/web-view/HOW-TO.html#webview

I didn't go deep into it but since I will need it very soon I really hope there is a way to **reuse the auth token obtained in the webview in the fully native part**(via view controllers) .

-----Original Message-----

**Comment by dragGH102** [comment]
@sitefinitysteve is Oauth finally fixed for Android in {n} 1.60 ?

**Comment by valentinstoychev** [comment]
It is not part of the official release, but I believe it is done in the master branch. I'll let @slavchev to confirm.

**Comment by enchev** [comment]
Hey @dragGH102 @sitefinitysteve,

Do you need anything more from the runtimes or I can close this?

**Comment by sitefinitysteve** [comment]
I'm still waiting on an Android fix :/ https://github.com/NativeScript/android-runtime/issues/546#issuecomment-246711994

**Comment by valentinstoychev** [comment]
@sitefinitysteve this is available now in 2.4.

**Comment by votiakov** [comment]
Hello! If someone is looking for a social login plugin - check this out: https://github.com/mkloubert/nativescript-social-login

It now has Google and Facebook login on both iOS and Android. All native.
Twitter will probably come next.
